waiter/src/waiter/state/responder.clj

```diff
@@ -482,9 +483,14 @@
   [{:keys [instance-id->state] :as current-state} update-instance-id->blacklist-expiry-time-fn update-status-tag-fn
    instance-id expiry-time]
   (log/info "unblacklisting instance" instance-id "as blacklist expired at" expiry-time)
-  (cond-> (update-instance-id->blacklist-expiry-time-fn current-state #(dissoc % instance-id))
-    (contains? instance-id->state instance-id)
-    (update-in [:instance-id->state instance-id] update-status-tag-fn #(disj % :blacklisted))))
```

Comment:
I'd prefer this same structure. We can wrap the `update-in` into a helper function that adds the necessary instance tracking logging. The `if-else` feels clunky here.

Reply:
Done